public/video-ui/src/components/ScheduledLaunch/ScheduledLaunch.js

```diff
@@ -11,89 +17,338 @@ export default class ScheduledLaunch extends React.Component {
+    this.setState({ selectedEmbargoDate: embargo || scheduledLaunch });
+  }
+
+  validateDate = (date, actionType) => {
+    if (!date) {
+      this.setState({ invalidDateError: null });
+      return;
+    }
+
+    const { video: { contentChangeDetails } } = this.props;
+    const scheduledLaunch =
+      contentChangeDetails &&
+      contentChangeDetails.scheduledLaunch &&
+      contentChangeDetails.scheduledLaunch.date;
+    const embargo =
+      contentChangeDetails &&
+      contentChangeDetails.embargo &&
+      contentChangeDetails.embargo.date;
+
+    this.setState({ invalidDateError: null });
+
+    if (!isFutureDate(date)) {
+      this.setState({ invalidDateError: 'Date must be in the future!' });
+    }
+
+    if (
+      actionType === 'schedule' &&
+      embargo &&
+      isFutureDate(date) &&
+      !isSameOrAfter(date, embargo)
+    ) {
+      this.setState({
+        invalidDateError: "Scheduled launch can't be earlier than embargo!"
+      });
+    }
+
+    if (
+      actionType === 'embargo' &&
+      scheduledLaunch &&
+      isFutureDate(date) &&
+      isAfter(date, scheduledLaunch)
+    ) {
+      this.setState({
+        invalidDateError: "Embargo can't be later than scheduled launch!"
+      });
+    }
   };
 
-  saveScheduledLaunch = () => {
+  onSelectOption = actionType => {
+    const date =
+      actionType === 'schedule'
+        ? this.state.selectedScheduleDate
+        : this.state.selectedEmbargoDate;
+    this.validateDate(date, actionType);
+    this.setState({
+      showDatePicker: true,
+      actionType,
+      showScheduleOptions: false
+    });
+  };
+
+  preventPublication = () => {
     const video = this.props.video;
-    this.props.saveVideo(Object.assign({}, video, {
+    const impossiblyDistantDate = 16725225600000;
+    this.props.saveVideo(
+      Object.assign({}, video, {
         contentChangeDetails: Object.assign({}, video.contentChangeDetails, {
-          scheduledLaunch: Object.assign({}, video.contentChangeDetails.scheduledLaunch, {
-            date: this.state.selectedDate
+          embargo: Object.assign({}, video.contentChangeDetails.embargo, {
+            date: impossiblyDistantDate
           })
         })
       })
     );
-    this.setState({ showDatePicker: false });
-  }
+    this.setState({
+      selectedEmbargoDate: impossiblyDistantDate,
+      showScheduleOptions: false
+    });
+  };
+
+  setDate = (date, actionType) => {
+    if (!actionType) return;
+    const key =
+      actionType === 'schedule'
+        ? 'selectedScheduleDate'
+        : 'selectedEmbargoDate';
 
-  removeScheduledLaunch = () => {
+    this.setState({ [key]: date });
+    this.validateDate(date, actionType);
+  };
+
+  saveDate = actionType => {
+    const isSchedule = actionType === 'schedule';
+    const key = isSchedule ? 'scheduledLaunch' : 'embargo';
+    const selectedDate =
+      actionType === 'schedule'
+        ? 'selectedScheduleDate'
+        : 'selectedEmbargoDate';
```

Comment:
Could we somehow reduce the amount of different variable names going on here? Instead of actionType, could we just have a string called propertyName and could the propertyName be also used to identify the selectedDate?